File: ts/src/task/containsDuplicate_217.ts
// export function containsDuplicate(nums: number[]): boolean {
//   let set = new Set<number>();
//   nums.forEach(function (n) {
//     if (set.has(n)) {
//       return true;
//     }
//     set.add(n);
//   });
//   return false;
// }

export function containsDuplicate(nums: number[]): boolean {
  let set = new Set<number>(nums);
  return set.size != nums.length ? true : false;
}

File: dart/lib/task/containsDuplicate_217.dart
// bool containsDuplicate(List<int> nums) {
//   Set<int> setNumber = nums.toSet();
//   return nums.length != setNumber.length ? true : false;
// }

bool containsDuplicate(List<int> nums) {
  Set<int> newSet = {};

  for (var n in nums) {
    if (newSet.contains(n)) {
      return true;
    }
    newSet.add(n);
  }
  return false;
}
